packages/botonic-nlp/src/preprocess/vocabulary-generator.ts

```diff
@@ -3,7 +3,11 @@ import { PADDING_TOKEN } from './constants'
 import { Preprocessor } from './preprocessor'
 
 export class VocabularyGenerator {
-  constructor(private preprocessor: Preprocessor) {}
+  private constructor(private preprocessor: Preprocessor) {}
+
+  static using(preprocessor: Preprocessor): VocabularyGenerator {
```

Comment:
factory method is used when we have several ways to construct the class or when you need async calls.
In this case we probably don't need it